nautobot/dcim/models/devices.py

```diff
@@ -1209,3 +1217,91 @@ def __str__(self):
         if self.alias:
             return self.alias
         return f"{self.platform} - {self.version}"
+
+
+#
+# Controller
+#
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "dynamic_groups",
+    "export_templates",
+    "graphql",
+    "locations",
+    "statuses",
+    "webhooks",
+)
+class Controller(PrimaryModel, ConfigContextModel):
+    """Controller model."""  # TODO: (whitej6) finialize description
+
+    name = models.CharField(max_length=255, unique=True)
+    status = StatusField(blank=False, null=False)
+    description = models.CharField(max_length=200, blank=True)
+    location = models.ForeignKey(
+        to="dcim.Location",
+        on_delete=models.PROTECT,
+        related_name="controllers",
+    )
+    platform = models.ForeignKey(
+        to="dcim.Platform",
+        on_delete=models.SET_NULL,
+        related_name="controllers",
+        blank=True,
+        null=True,
+    )
+    role = RoleField(blank=False, null=False)
+    tenant = models.ForeignKey(
+        to="tenancy.Tenant",
+        on_delete=models.PROTECT,
+        related_name="controllers",
+        blank=True,
+        null=True,
+    )
+    external_integration = models.ForeignKey(
+        to="extras.ExternalIntegration",
+        on_delete=models.SET_NULL,
+        related_name="controllers",
+        blank=True,
+        null=True,
+    )
+    deployed_controller_device = models.ForeignKey(
+        to="dcim.Device",
+        on_delete=models.SET_NULL,
+        related_name="controllers",
+        blank=True,
+        null=True,
+    )
+    deployed_controller_group = models.ForeignKey(
+        to="dcim.DeviceRedundancyGroup",
+        on_delete=models.SET_NULL,
+        related_name="controllers",
+        blank=True,
+        null=True,
+    )
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "dynamic_groups",
+    "export_templates",
+    "graphql",
+    "locations",
+    "statuses",
+    "webhooks",
+)
+class ControllerDeviceGroup(TreeModel, PrimaryModel, ConfigContextModel):
+    """Controller model."""  # TODO: (whitej6) finialize description
+
+    name = models.CharField(max_length=255, unique=True)
+    weight = models.PositiveIntegerField(default=1000)
+    controller = models.ForeignKey(
+        to="dcim.Controller",
+        on_delete=models.CASCADE,
+        related_name="+",
```

Comment:
We shouldn't be using `related_name="+"` unless there's a *strong* justification for it.

Reply:
Fixed